src/sidebar/components/SidebarView.tsx

```diff
@@ -158,7 +159,13 @@ function SidebarView({
           <PendingUpdatesNotification />
         </div>
       )}
-      <ThreadList threads={rootThread.children} />
+      <div
+        role="tabpanel"
+        id={`${selectedTab}-panel`}
+        aria-labelledby={`${selectedTab}-tab`}
+      >
```

Comment:
I see a few issues here:

1. The association will be broken if `showTabs` is false, as the tabs are not rendered in that case.
2. The element ID here and in `SelectionTabs` are generated separately, which makes it a bit easier for a refactoring to accidentally break them. It would be better if we could avoid that somehow (eg. by generating the values once and passing them down, or using a shared helper to generate them).
3. The `SelectionTabs` component is not just the tab headers, but it also contains fallback content for the tabs in several cases. This fallback content needs to be associated with the tab headers somehow.

I suspect these issues might be easier to solve with some refactoring of the component structure.

Reply:
As part of the search UI refactoring we reduced the number of states where `showTabs` is false. I wonder if we could eliminate that state entirely so the tab bar is always shown.

Reply:
> I see a few issues here:
> 
> 1. The association will be broken if `showTabs` is false, as the tabs are not rendered in that case.
> 2. The element ID here and in `SelectionTabs` are generated separately, which makes it a bit easier for a refactoring to accidentally break them. It would be better if we could avoid that somehow (eg. by generating the values once and passing them down, or using a shared helper to generate them).
> 3. The `SelectionTabs` component is not just the tab headers, but it also contains fallback content for the tabs in several cases. This fallback content needs to be associated with the tab headers somehow.
> 
> I suspect these issues might be easier to solve with some refactoring of the component structure.

Ok, this is what I think I'm going to do to address these issues.

Right now we have two components representing tabs and tab content

* `SelectionTabs` -> renders the tabs, and in some cases part of the tab panel fallback content.
* `ThreadList` -> renders the rest of the tab panel content.

I'm going to rename `SelectionTabs`, to something like `SidebarTabsPanel`, and allow it to receive `children`. Then, I will pass `ThreadList` as the `SidebarTabsPanel`, so that it is in control of where is it rendered.

Then I will generate the tab panel "wrapper" inside the new `SidebarTabsPanel`, making sure it includes the conditional pieces of content that are currently in `SelectionTabs`, followed by `children`.

By doing this, the tab/tab panel ID generation can be defined in a single place to make sure there's no possible mistakes.

The only case left is when `showTabs` is false. In that case we are currently not rendering `SelectionTabs`, but we do render `ThreadList`. However, I think `ThreadList` is always incidentally empty in case of error (which is what determines if `showTabs` is `true` or `false`), so it should be safe to render it only in case of tabs.

I need to fully verify this last point though.

Reply:
> However, I think ThreadList is always incidentally empty in case of error (which is what determines if showTabs is true or false), 

It looks like the case when this error occurs is when a direct-linked annotation or group cannot be found. Can you verify what happens in that case?

Reply:
This plan sounds reasonable to me. Another option would be to have the tab component also handle fetching the root thread by calling `useRootThread`, then render the `ThreadList` itself. I couldn't be sure which will work out best without trying it.

Reply:
> Another option would be to have the tab component also handle fetching the root thread by calling useRootThread, then render the ThreadList itself. I couldn't be sure which will work out best without trying it.

Actually, I think I like this better 😅

Reply:
> > However, I think ThreadList is always incidentally empty in case of error (which is what determines if showTabs is true or false),
> 
> It looks like the case when this error occurs is when a direct-linked annotation or group cannot be found. Can you verify what happens in that case?

I have tested this, and I see we render no annotations when linking to a not-found annotation.

![image](https://github.com/hypothesis/client/assets/2719332/e5b1addf-817a-4eec-866d-018aeb669a2a)

However, we do seem to be loading all annotations (fro default group it seems) when trying to link to a group that does not exist

![image](https://github.com/hypothesis/client/assets/2719332/3a4c843d-35f3-4d86-b8d3-c4f89222b02e)

So I would need to adapt the logic to continue rendering the `ThreadList` and skip only the tabs, when using the new approach discussed above.